rust/main/hyperlane-base/src/traits/checkpoint_syncer.rs

```diff
@@ -14,9 +14,15 @@ pub trait CheckpointSyncer: Debug + Send + Sync {
     async fn write_latest_index(&self, index: u32) -> Result<()>;
     /// Update the latest index of this syncer if necessary
     async fn update_latest_index(&self, index: u32) -> Result<()> {
-        let curr = self.latest_index().await?.unwrap_or(0);
-        if index > curr {
-            self.write_latest_index(index).await?;
```

Comment:
I wonder if we use `-1` as default, this code will work as needed?

Reply:
it would! Basically the current solution uses `None` as `-1`, which I think is more idiomatic to rust

Reply:
yeah agree with @daniel-savu . if we use -1, we will have to do a bunch of casting to `i64` and back